# Step 1 → Understand the Business Problem + Why Confidence Intervals Matter

In [ ]:
# Step 1: Business Understanding

# A manufacturer wants to estimate the average durability
# of print-heads before failure.

# Since testing is destructive and expensive,
# only a small sample of 15 print-heads was tested.

# Confidence intervals help estimate the true population mean
# using sample data.

# We will construct:
# 1. 99% Confidence Interval using sample standard deviation (t-distribution)
# 2. 99% Confidence Interval using known population standard deviation (z-distribution)

# Step 2 — Organize and Understand the Dataset

In [ ]:
# Step 2: Organize Dataset

import numpy as np
import pandas as pd

# Durability data (in millions of characters)

data = [
    1.13, 1.55, 1.43, 0.92, 1.25,
    1.36, 1.32, 0.85, 1.07, 1.48,
    1.20, 1.33, 1.18, 1.22, 1.29
]

# Convert into DataFrame

df = pd.DataFrame(data, columns=["Durability"])

# Display dataset

print(df)

    Durability
0         1.13
1         1.55
2         1.43
3         0.92
4         1.25
5         1.36
6         1.32
7         0.85
8         1.07
9         1.48
10        1.20
11        1.33
12        1.18
13        1.22
14        1.29


## Initial Dataset Understanding

We observe that:

**Values mostly lie between:**

**0.85 to 1.55**

million characters

This suggests:

-- moderate spread

-- no extremely impossible values

-- reasonable durability consistency


Good for quality estimation.

In [ ]:
# Sample Size

n = len(data)

print("Sample Size:", n)

Sample Size: 15


In [ ]:
# Since sample size is small (n < 30),
# t-distribution becomes important when population
# standard deviation is unknown.

In **manufacturing and quality control** :

small samples are common when testing is :

-- destructive

-- expensive

-- time-consuming

So analysts must estimate population characteristics carefully using limited data.

This is real-world industrial statistics.

# Step 3 — Calculate Sample Mean and Sample Standard Deviation

In [ ]:
# Sample Mean

sample_mean = np.mean(data)

# Sample Standard Deviation

sample_std = np.std(data, ddof=1)

# ddof=1 is used because we are calculating
# sample standard deviation, not population standard deviation.

print("Sample Mean:", round(sample_mean, 4))
print("Sample Standard Deviation:", round(sample_std, 4))

Sample Mean: 1.2387
Sample Standard Deviation: 0.1932


## Interpretation
**Sample Mean**

xˉ ≈ 1.238

This means:

Average print-head durability is approximately:

**1.238 million characters**

before failure.

**Sample Standard Deviation**

s ≈ 0.193

This indicates:

Moderate variation exists in durability across print-heads.

This is important for quality consistency analysis.

## Real Analyst Thinking

Manufacturers care about:

**consistency**

not only average performance.

Two products may have same mean durability,

but:

**lower standard deviation**

usually means:

-- better manufacturing quality

-- higher reliability

-- fewer defective products

This is real-world quality analytics.

Very strong insight.

# Step 4 — Build 99% Confidence Interval using t-Distribution

In [ ]:
# Step 4: 99% Confidence Interval using t-Distribution

from scipy import stats
import math

# Sample statistics

sample_mean = np.mean(data)
sample_std = np.std(data, ddof=1)
n = len(data)

# Confidence level

confidence_level = 0.99

# Degrees of freedom

df = n - 1

# Critical t-value

t_critical = stats.t.ppf(1 - 0.005, df)

# Margin of Error

margin_error = t_critical * (sample_std / math.sqrt(n))

# Confidence Interval

lower_limit = sample_mean - margin_error
upper_limit = sample_mean + margin_error

print("Sample Mean:", round(sample_mean, 4))
print("Sample Standard Deviation:", round(sample_std, 4))
print("t-Critical Value:", round(t_critical, 4))

print("\n99% Confidence Interval:")
print((round(lower_limit, 4), round(upper_limit, 4)))

Sample Mean: 1.2387
Sample Standard Deviation: 0.1932
t-Critical Value: 2.9768

99% Confidence Interval:
(np.float64(1.0902), np.float64(1.3871))


### Interpretation

The 99% confidence interval suggests that:

**The true mean durability of print-heads likely lies between:**

1.0902 and 1.3871

million characters before failure.

This gives the manufacturer:

--- reliability estimate

--- uncertainty range

--- quality assessment support

### Real Analyst Thinking

Manufacturers do not only want:

**average durability**

They also want:

**reliability confidence**

Confidence intervals help businesses understand:

-- uncertainty

-- risk

-- expected performance range

-- manufacturing consistency

This is real-world quality control analytics.

Very valuable.

# Step 5 — Explain Why t-Distribution is Used

In [ ]:
# t-distribution is used because:
# 1. Sample size is small (n = 15)
# 2. Population standard deviation is unknown

# Small samples create additional uncertainty,
# and t-distribution accounts for this uncertainty
# using heavier tails and wider confidence intervals.

# As sample size increases,
# t-distribution approaches the normal distribution.

### Real Analyst Thinking

In real manufacturing and quality control:

businesses often cannot collect large samples because:

-- testing is expensive

-- testing is destructive

-- production batches are limited

So analysts must make reliable estimates from small samples.

The t-distribution makes those estimates safer and more realistic.

# Step 6 — Build 99% Confidence Interval using Known Population Standard Deviation (Z-Interval)

In [ ]:
from scipy import stats
import math

# Known population standard deviation

sigma = 0.2

# Sample statistics

sample_mean = np.mean(data)
n = len(data)

# Critical z-value for 99% confidence

z_critical = stats.norm.ppf(1 - 0.005)

# Margin of Error

margin_error = z_critical * (sigma / math.sqrt(n))

# Confidence Interval

lower_limit = sample_mean - margin_error
upper_limit = sample_mean + margin_error

print("Sample Mean:", round(sample_mean, 4))
print("Population Standard Deviation:", sigma)
print("z-Critical Value:", round(z_critical, 4))

print("\n99% Confidence Interval:")
print((round(lower_limit, 4), round(upper_limit, 4)))

Sample Mean: 1.2387
Population Standard Deviation: 0.2
z-Critical Value: 2.5758

99% Confidence Interval:
(np.float64(1.1057), np.float64(1.3717))


### Interpretation

The 99% z-confidence interval suggests that:

**The true mean durability of print-heads likely lies between:**

1.1057 and 1.3717

million characters before failure.

This interval is slightly narrower than the t-interval because:

**known population standard deviation reduces uncertainty.**

### Real Analyst Thinking

Businesses prefer:

**narrower confidence intervals**

because they provide:

-- more precise estimates

-- lower uncertainty

-- stronger quality predictions

Knowing population variability improves reliability estimation significantly.

# Step 7 — Compare t-Interval vs z-Interval

The confidence interval constructed using the t-distribution is wider than the z-distribution interval because the population standard deviation is unknown and must be estimated using sample data.

This estimation introduces additional uncertainty, which is accounted for by the t-distribution through larger critical values and heavier tails.

When the population standard deviation is known, uncertainty decreases, resulting in a narrower and more precise z-confidence interval.

| Feature                       | t-Distribution    | z-Distribution |
| ----------------------------- | ----------------- | -------------- |
| Population Standard Deviation | Unknown           | Known          |
| Sample Size                   | Small             | Any            |
| Critical Value                | Larger            | Smaller        |
| Interval Width                | Wider             | Narrower       |
| Uncertainty                   | Higher            | Lower          |
| Reliability                   | More Conservative | More Precise   |


### Real Analyst Thinking

In real-world quality control:

Businesses prefer:

**precise estimates**

but must also account for:

**uncertainty and risk**

When uncertainty is higher:

-- safer estimates are preferred

This is why:

**t-distribution is more conservative**

and commonly used in small-sample industrial studies.


# Step 8 — Final Business Conclusion + Professional Summary

# Final Conclusion

In this project, confidence interval estimation was performed to estimate the average durability of print-heads manufactured for personal computers.

Since the testing process was destructive in nature, only a small sample of:

n = 15

print-heads could be tested before failure. The durability values were measured in millions of characters printed before failure.

The sample mean durability was calculated as:

xˉ ≈ 1.238

million characters,

while the sample standard deviation was approximately:

s ≈ 0.193

million characters.

Two different 99% confidence intervals were constructed:

### **1. Confidence Interval using t-Distribution**

Since the sample size was small and the population standard deviation was unknown, the t-distribution was used.

The resulting 99% confidence interval was approximately:

(1.0902,1.3871)

million characters.

### **2. Confidence Interval using z-Distribution**

When the population standard deviation was assumed to be known:

σ = 0.2

million characters,

the z-distribution was used.

The resulting 99% confidence interval was approximately:

(1.1057,1.3717)

million characters.

### **Key Statistical Insight**

The t-confidence interval was slightly wider than the z-confidence interval because estimating population variability from a small sample introduces additional uncertainty.

The t-distribution accounts for this uncertainty using larger critical values and heavier tails, making it more appropriate for small-sample inference problems.

### **Business Interpretation**

These confidence intervals provide the manufacturer with an estimated range in which the true average durability of print-heads likely lies.

This information is extremely valuable for:

-- quality control

-- reliability estimation

-- warranty planning

-- production monitoring

-- manufacturing consistency evaluation

Instead of relying only on a single sample average, confidence intervals provide a more reliable understanding of uncertainty and expected product performance.

This helps businesses make safer and more informed operational decisions.

### **Final Analytical Insight**

This project demonstrates how statistical estimation techniques help industries make reliable conclusions even when only limited sample data is available.

In real-world manufacturing environments where destructive testing is expensive, confidence intervals become essential tools for balancing product quality assurance with operational cost efficiency.

Thus, this project highlights the practical importance of statistical inference, uncertainty estimation, and evidence-based quality control in industrial analytics.